packages/atom/src/sliceable.ts

```diff
@@ -1,34 +1,115 @@
-import { Slice } from './slice';
+import { Subscribable } from '@effection/subscription';
+import { Operation } from 'effection';
 
-export type Sliceable<S, A> = {
+export interface Slice<S> extends Subscribable<S,undefined> {
+  get(): S;
+  set(value: S): void;
+  update(fn: (state: S) => S): void;
+  slice(): Sliceable<S>;
+  once(predicate: (state: S) => boolean): Operation<S>;
+  remove(): void;
+  over(fn: (value: S) => S): void;
+}
+export type Atom<S> = Omit<Slice<S>, 'remove' | 'over'> & {
+  reset(initializer?: (initial: S, curr: S) => S): void;
+  setMaxListeners(value: number): void;
```

Comment:
I'm not a huge fan of `reset` anyway, it's a symptom of problems in bigtest that we're trying to hack around. As for `setMaxListeners`, we could just set that number very high and hope for the best 🙈

Reply:
@jnicklas @cowboyd I agree about `setMaxListeners`. I only ever remember seeing an error about this when I had a memory leak with an EventEmitter and too many were getting created. 

I think the code can really be simplified more but not for this PR, I wanted to keep it as close to the original API as possible but I think the code surface area could be quite small.

I think I can get rid of the horrible overloads with typescript 4.1.2 also if we upgrade.

Reply:
@cowboyd @jnicklas I sort of feel there is unnecessary duplication in this current PR, I think the `createAtom` function could just create a `Slice` using the `sliceMaker` HOF and have 1 set of `get`, `set` etc .

It would be exactly the same API just get rid of the duplication that is nagging at me.

Maybe that is one for a future PR or maybe....just go with this?